classes/03class/aws/README.md

```diff
@@ -0,0 +1,184 @@
+### Automating EC2 instances
+
+When thinking how to design an application/server to run from a dynamic instance, you will have to think if the required software change often or not.
+
+
+#### AMI
+
+For example, my server needs an antivirus agent, nginx webserver and once it starts, it needs to deploy the application itself.
+
+Both the antivirus agent and the nginx webserver won't change their versions very often (less than once per week), so it will be a good idea to create a custom [Amazon Machine Image (AMI)](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) with those softwares already installed. So when the ASG creates a new instance, those software will be there already.
+
+#### UserData
+
+The application deployment gets a little bit trickier as you will expect that everytime a new server comes up, to go and pickup your latest code to be deployed. So this is not a good candidate to be part of the instance image.
+
+The UserData is a script (bash or other languages) that you can make it run once your EC2 instance have the operational system ready to start.
+
+On this script you can for example, checkout a git repository, build your code and deploy to your running nginx webserver.
+
+
+## AWS Load Balancer
+
+Your webserver is now running on an ASG and being scaled up and down based on your computing resources needs or any other metrics that you chose. How would your users know to which instance they need to connect and how many of those are available at any point in time?
+
+That's how the load balancer works. Your clients will access through the load balancer and the LB will be in charge of balancing the requests across your instances.
+
+![ASG-LB](/classes/03class/aws/assets/ASG-LB.png)
+
+Load balancers can be of a few different types depending on what layer the routing decision should be:
+
+- Application Load Balancer
+  - application layer (HTTP/HTTPS)
+- Network Load Balancer
+  - transport layer (TCP/SSL)
+- Classic Load Balancer
+  - either application (HTTP/HTTPS) or transport (TCP/SSL) layer
+
+This material will be focused only on the Application Load Balancer which is used for most of the use cases.
+
+### Application Load Balancer (ALB)
+
+The service can be found at Services > Compute > EC2 > Load Balancing > Load Balancers > Create Load Balancer > ALB
+
+An ALB can serve as a single point of contact for clients and distribute traffic across multiple targets, such as EC2 Instances, in multiple Availability Zones. When creating an ALB you will have to chose from 2 different types and in which Availability Zones your LB will be available:
+
+- internet-facing: routes requests from public internet to targets
+- internal: routes requests only from clients to targets using private IP addresses
+
+Basically an ALB has two main components: listeners and target groups.
+
+```Load balancers are included on the AWS free-tier. However, please remember to delete any unused resources to avoid any costs```
+
+#### listeners
```

Comment:
ANy reason to have those sections lower case?

Reply:
it's just being used as bold, no need to be a section on the appendix